geoutils/georaster/raster.py

```diff
@@ -958,12 +1002,101 @@ def copy(self: RasterType, new_array: np.ndarray | None = None) -> RasterType:
+                        "transform": self.transform,
+                        "crs": self.crs,
+                        "nodata": self.nodata,
+                    }
+                )
+
+            # If the universal function has two outputs (Note: no ufunc exists that has three outputs or more)
+            else:
+                output = getattr(ufunc, method)(inputs[0].data, **kwargs)  # type: ignore
+                return self.__class__(
+                    {"data": output[0], "transform": self.transform, "crs": self.crs, "nodata": self.nodata}
+                ), self.__class__(
+                    {"data": output[1], "transform": self.transform, "crs": self.crs, "nodata": self.nodata}
+                )
+
+        # If the universal function takes two inputs (Note: no ufunc exists that has three inputs or more)
+        else:
+            if ufunc.nout == 1:
+                return self.__class__(
+                    {
+                        "data": getattr(ufunc, method)(inputs[0].data, inputs[1].data, **kwargs),  # type: ignore
+                        "transform": self.transform,
+                        "crs": self.crs,
+                        "nodata": self.nodata,
+                    }
+                )
+
+            # If the universal function has two outputs (Note: no ufunc exists that has three outputs or more)
+            else:
+                output = getattr(ufunc, method)(inputs[0].data, inputs[1].data, **kwargs)  # type: ignore
+                return self.__class__(
+                    {"data": output[0], "transform": self.transform, "crs": self.crs, "nodata": self.nodata}
+                ), self.__class__(
+                    {"data": output[1], "transform": self.transform, "crs": self.crs, "nodata": self.nodata}
+                )
+
+    def __array_function__(self, func: Callable[[np.ndarray, Any], Any], types: type, args: Any, kwargs: Any) -> Any:
+        """
+        Method to cast NumPy array function directly on a Raster object by applying it to the masked array.
+        A limited number of function is supported, listed in raster._HANDLED_FUNCTIONS.
+        """
+
+        # If function is not implemented
+        if func.__name__ not in _HANDLED_FUNCTIONS:
+            return NotImplemented
+
+        # For subclassing
+        if not all(issubclass(t, self.__class__) for t in types):
+            return NotImplemented
+
+        # We now choose the behaviour of array functions
+        # For median, np.median ignores masks of masked array, so we force np.ma.median
+        if func.__name__ in ["median", "nanmedian"]:
+            func = np.ma.median
+            first_arg = args[0].data
+
+        # For percentiles and quantiles, there exist no masked array version, so we compute on the valid data directly
+        elif func.__name__ in ["percentile", "nanpercentile"]:
+            first_arg = args[0].data.data[~args[0].data.mask]
```

Comment:
You can use `self.data.compressed()` to get only the valid values (I know, the name is not very straightforward...).

Reply:
Good point, changed everywhere!